suzieq/poller/sq_poller.py

```diff
@@ -206,20 +206,36 @@ async def start_poller(userargs, cfg):
 
     ignore_known_hosts = userargs.ignore_known_hosts or None
 
+    # Retrieve the services to start
+    svcs = list(Path(cfg["service-directory"]).glob('*.yml'))
+    allsvcs = [os.path.basename(x).split('.')[0] for x in svcs]
+    svclist = None
+
     if userargs.service_only:
         svclist = userargs.service_only.split()
+
+        # Check if all the given services are valid
+        notvalid = [s for s in svclist if s not in allsvcs]
+        if notvalid:
+            print(f'Invalid sevices specified: {notvalid}. '
+                  f'Should have been one of {allsvcs}')
+            exit(1)
     else:
-        svcs = list(Path(cfg["service-directory"]).glob('*.yml'))
-        svclist = [os.path.basename(x).split('.')[0] for x in svcs]
+        svclist = allsvcs
 
     if userargs.exclude_services:
-        svclist = list(filter(lambda x: x not in userargs.exclude_services,
+        excluded_services = userargs.exclude_services.split()
+        # Check if all the excluded services are valid
+        notvalid = [e for e in excluded_services if e not in allsvcs]
+        if notvalid:
+            print(f'Services {notvalid} excluded, but they '
```

Comment:
Why should we continue in this case? For example, if the user typed iterfaces and we continued, that'd be not good, especially if we're not doing so in the case of the other option, "-s". I think you should terminate this as well.